core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1543,24 +1543,34 @@ class KafkaApis(val requestChannel: RequestChannel,
     var skippedMarkers = 0
     for (marker <- markers.asScala) {
       val producerId = marker.producerId
-      val (goodPartitions, partitionsWithIncorrectMessageFormat) = marker.partitions.asScala.partition { partition =>
+      val partitionsWithCorrectMessageFormat = new mutable.ListBuffer[TopicPartition]
+      val partitionsWithIncorrectMessageFormat = new mutable.ListBuffer[TopicPartition]
+      val unknownPartitions = new mutable.ListBuffer[TopicPartition]
```

Comment:
Nit: `ArrayBuffer` is generally preferred for mutable sequences (similar to `ArrayList` versus `LinkedList` in Java.).